File: db/import_csvs.js
const fs = require("fs");
const path = require("path");
const sqlite3 = require("sqlite3").verbose();
const csv = require("csv-parser");

const db = new sqlite3.Database("./data.db");
const csvDir = "./data";

function importCSV(filePath, tableName) {
  return new Promise((resolve, reject) => {
    const rows = [];

    console.log(`Reading file: ${filePath}`);

    fs.createReadStream(filePath)
      .pipe(csv())
      .on("data", (data) => rows.push(data))
      .on("end", () => {
        if (rows.length === 0) {
          console.warn(`⚠️ Skipping empty file: ${tableName}`);
          return resolve();
        }

        const columns = Object.keys(rows[0]);

        db.serialize(() => {
          db.run(`DROP TABLE IF EXISTS "${tableName}"`, (err) => {
            if (err) return reject(err);
            const createStmt = `CREATE TABLE "${tableName}" (${columns
              .map((c) => `"${c}" TEXT`)
              .join(", ")})`;

            db.run(createStmt, (err) => {
              if (err) return reject(err);

              const insertStmt = db.prepare(
                `INSERT INTO "${tableName}" (${columns
                  .map((c) => `"${c}"`)
                  .join(", ")}) VALUES (${columns.map(() => "?").join(", ")})`
              );

              for (const row of rows) {
                insertStmt.run(columns.map((c) => row[c]));
              }

              insertStmt.finalize((err) => {
                if (err) return reject(err);
                console.log(`✅ Imported table: ${tableName}`);
                resolve();
              });
            });
          });
        });
      })
      .on("error", (err) => {
        console.error(`❌ Error reading ${tableName}:`, err);
        reject(err);
      });
  });
}

async function runImport() {
  const files = fs.readdirSync(csvDir).filter((f) => f.endsWith(".csv"));

  for (const file of files) {
    const filePath = path.join(csvDir, file);
    const tableName = path.basename(file, ".csv");

    try {
      await importCSV(filePath, tableName);
    } catch (err) {
      console.error(`❌ Failed to import ${file}:`, err);
    }
  }

  db.close(() => {
    console.log("📦 All imports complete. Database closed.");
  });
}

runImport();

File: db/import_csvs.py
import os
import pandas as pd
import sqlite3

# Path to your CSV folder and output DB file
csv_dir = './dc_data'
sqlite_db = './dc_data.db'

# Connect to SQLite database (it'll be created if it doesn't exist)
conn = sqlite3.connect(sqlite_db)

# Loop through all CSV files
for filename in os.listdir(csv_dir):
    if filename.endswith('.csv'):
        table_name = os.path.splitext(filename)[0]
        file_path = os.path.join(csv_dir, filename)

        print(f'📥 Importing {filename} into table: {table_name}')

        try:
            # Load the CSV into a pandas DataFrame
            df = pd.read_csv(file_path, dtype=str).fillna('')  # keep all as text for safety

            print(f'→ {len(df)} rows, columns: {list(df.columns)}')

            # Import to SQLite
            df.to_sql(table_name, conn, if_exists='replace', index=False)
            print(f'✅ Imported table: {table_name}\n')
        except Exception as e:
            print(f'❌ Failed to import {filename}: {e}\n')

# Done
conn.close()
print("🏁 All imports complete. Database saved to:", sqlite_db)
